Library/Homebrew/cask/lib/hbc/installer.rb

```diff
@@ -83,7 +84,7 @@ def install
       odebug "Hbc::Installer#install"
 
       if @cask.installed? && !force? && !@reinstall
-        raise CaskAlreadyInstalledError, @cask
+        raise CaskAlreadyInstalledError, @cask unless upgrade?
```

Comment:
Add `!upgrade?` to the `if` above instead, or move `force? || @reinstall` down here.